tests/functional/provider_client/test_onboarding_token_generation.py

```diff
@@ -0,0 +1,103 @@
+import logging
+import pytest
+import time
+
+from ocs_ci.ocs.resources import pod
+from ocs_ci.ocs import constants, ocp
+from ocs_ci.framework import config
+from ocs_ci.framework.testlib import (
+    skipif_ocs_version,
+    ManageTest,
+    tier1,
+    skipif_ocp_version,
+    skipif_managed_service,
+    runs_on_provider,
+    skipif_external_mode,
+    green_squad,
+)
+from ocs_ci.ocs.ui.validation_ui import ValidationUI
+
+log = logging.getLogger(__name__)
+
+
+@tier1
+@green_squad
+@skipif_ocs_version("<4.15")
+@skipif_ocp_version("<4.15")
+@skipif_external_mode
+@runs_on_provider
+@skipif_managed_service
+class TestOnboardingTokenGeneration(ManageTest):
+    @pytest.fixture(autouse=True)
+    def setup(self, request):
+        """
+        Resetting the default value of KeyRotation
+        """
+        self.pod_obj = ocp.OCP(
+            kind="Pod", namespace=config.ENV_DATA["cluster_namespace"]
+        )
+
+    def test_ux_server_pod_is_running(self):
+        """
+        Test to verify that ux-backend-server pod is up and running.
+
+        """
+        assert self.pod_obj.wait_for_resource(
+            condition=constants.STATUS_RUNNING,
+            selector=constants.UX_BACKEND_SERVER_LABEL,
+            resource_count=1,
+            timeout=180,
+        ), "ux pod is not in running state as expected"
+
+    def test_respin_of_ux_server_pod(self):
```

Comment:
I think a separate test case to check the re-spin of pod is not needed. This will be part of tier4c pod re-spin tests. Please combine the test with test_onboarding_token_generation_option_is_available_in_ui(and change the name of test) to verify the impact of ux pod respin on token generation.

Reply:
I have combined the test to check ux-backend-server pod is urunning and ux-backend-server pod respin testcases together.